sentry-android-replay/src/main/java/io/sentry/android/replay/ReplayCache.kt

```diff
@@ -237,9 +240,184 @@ public class ReplayCache internal constructor(
+            val width = lastSegment[SEGMENT_KEY_WIDTH]?.toIntOrNull()
+            val frameRate = lastSegment[SEGMENT_KEY_FRAME_RATE]?.toIntOrNull()
+            val bitRate = lastSegment[SEGMENT_KEY_BIT_RATE]?.toIntOrNull()
+            val segmentId = lastSegment[SEGMENT_KEY_ID]?.toIntOrNull()
+            val segmentTimestamp = try {
+                DateUtils.getDateTime(lastSegment[SEGMENT_KEY_TIMESTAMP].orEmpty())
+            } catch (e: Throwable) {
+                null
+            }
+            val replayType = try {
+                ReplayType.valueOf(lastSegment[SEGMENT_KEY_REPLAY_TYPE].orEmpty())
+            } catch (e: Throwable) {
+                null
+            }
+            if (height == null || width == null || frameRate == null || bitRate == null ||
+                (segmentId == null || segmentId == -1) || segmentTimestamp == null || replayType == null
+            ) {
+                options.logger.log(
+                    DEBUG,
+                    "Incorrect segment values found for replay: %s, deleting the replay",
+                    replayId
+                )
+                FileUtils.deleteRecursively(replayCacheDir)
+                return null
+            }
+
+            val recorderConfig = ScreenshotRecorderConfig(
+                recordingHeight = height,
+                recordingWidth = width,
+                frameRate = frameRate,
+                bitRate = bitRate,
+                // these are not used for already captured frames, so we just hardcode them
+                scaleFactorX = 1.0f,
+                scaleFactorY = 1.0f
+            )
+
+            val cache = replayCacheProvider?.invoke(replayId, recorderConfig) ?: ReplayCache(options, replayId, recorderConfig)
+            cache.replayCacheDir?.listFiles { dir, name ->
+                if (name.endsWith(".jpg")) {
+                    val file = File(dir, name)
+                    val timestamp = file.nameWithoutExtension.toLongOrNull()
+                    if (timestamp != null) {
+                        cache.addFrame(file, timestamp)
+                    }
+                }
+                false
+            }
+
+            if (cache.frames.isEmpty()) {
+                options.logger.log(
+                    DEBUG,
+                    "No frames found for replay: %s, deleting the replay",
+                    replayId
+                )
+                FileUtils.deleteRecursively(replayCacheDir)
+                return null
+            }
+
+            cache.frames.sortBy { it.timestamp }
+
+            fun roundToNearestFrame(duration: Long, frameDuration: Int): Long {
+                val frames = duration.toDouble() / frameDuration.toDouble()
+                return ceil(frames).toLong() * frameDuration
+            }
+
+            // we need to round to the nearest frame to include breadcrumbs/events happened after the frame was captured
+            val duration = roundToNearestFrame(
```

Comment:
```suggestion
            val durationMillis = roundToNearestFrame(
```

Reply:
will address in the other PR since this line has changed there